From 2b02e60005cc00234d40899c8a1658261ad4a620 Mon Sep 17 00:00:00 2001
From: Oleksii Beketov <ol.beketov@samsung.com>
Date: Tue, 18 Dec 2018 14:21:07 +0200
Subject: [PATCH] Remove SHA384

1. MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384 ciphersuite removed
2. MBEDTLS_SHA512_C removed

Change-Id: I8d35aaecc6172c98238c5cd1450a3523f83f3776
Signed-off-by: Oleksii Beketov <ol.beketov@samsung.com>
---
 patches/08_mbedtls-reduce.patch | 19 +++++++++----------
 security/oc_tls.c               |  8 ++------
 2 files changed, 11 insertions(+), 16 deletions(-)

diff --git a/patches/08_mbedtls-reduce.patch b/patches/08_mbedtls-reduce.patch
index 347f6dd2..e2a819ac 100644
--- a/patches/08_mbedtls-reduce.patch
+++ b/patches/08_mbedtls-reduce.patch
@@ -1,6 +1,6 @@
-From 102ef9500a3d1d3a21c68b8e553fbd8c46d31db5 Mon Sep 17 00:00:00 2001
+From e1a2ac221fa366d768e34f81022d68389b47a2fa Mon Sep 17 00:00:00 2001
 From: Oleksii Beketov <ol.beketov@samsung.com>
-Date: Thu, 8 Nov 2018 14:11:09 +0200
+Date: Tue, 18 Dec 2018 14:18:06 +0200
 Subject: [PATCH] Reduce MbedTLS
 
 Reduce MbedTLS routines due memory optimization
@@ -9,7 +9,7 @@ Signed-off-by: Oleksii Beketov <ol.beketov@samsung.com>
 Signed-off-by: Aleksey Volkov <a.volkov@samsung.com>
 Signed-off-by: Oleksii Beketov <ol.beketov@samsung.com>
 ---
- include/mbedtls/config.h | 20 ++++++++++++++++----
+ include/mbedtls/config.h | 19 +++++++++++++++----
  include/mbedtls/sha512.h | 18 ------------------
  library/asn1parse.c      | 18 ++++++++++++++++++
  library/bignum.c         | 45 ++++++++++++++++++++++++++++++++++++++++++---
@@ -17,13 +17,13 @@ Signed-off-by: Oleksii Beketov <ol.beketov@samsung.com>
  library/pk.c             | 20 ++++++++++++++++++++
  library/pkparse.c        | 20 ++++++++++++++++++++
  library/ssl_cli.c        | 15 +++++++++++++++
- 8 files changed, 151 insertions(+), 25 deletions(-)
+ 8 files changed, 150 insertions(+), 25 deletions(-)
 
 diff --git a/include/mbedtls/config.h b/include/mbedtls/config.h
-index d9e5427..f791c54 100644
+index ec00023..9613189 100644
 --- a/include/mbedtls/config.h
 +++ b/include/mbedtls/config.h
-@@ -51,17 +51,26 @@
+@@ -50,8 +50,6 @@
  /* mbed TLS modules */
  #define MBEDTLS_AES_C
  #define MBEDTLS_CIPHER_C
@@ -31,9 +31,8 @@ index d9e5427..f791c54 100644
 -#define MBEDTLS_ENTROPY_C
  #define MBEDTLS_MD_C
  #define MBEDTLS_SHA256_C
-+#define MBEDTLS_SHA512_C
  #define MBEDTLS_SSL_COOKIE_C
- #define MBEDTLS_SSL_CLI_C
+@@ -59,8 +57,18 @@
  #define MBEDTLS_SSL_SRV_C
  #define MBEDTLS_SSL_TLS_C
  
@@ -52,7 +51,7 @@ index d9e5427..f791c54 100644
  
  /* Save some RAM by adjusting to your exact needs */
  #define MBEDTLS_PSK_MAX_LEN    16 /* 128-bits keys are generally enough */
-@@ -84,12 +93,13 @@
+@@ -83,12 +91,13 @@
  #define MBEDTLS_BIGNUM_C
  #define MBEDTLS_KEY_EXCHANGE_ECDH_ANON_ENABLED
  #define MBEDTLS_ECDH_C
@@ -68,7 +67,7 @@ index d9e5427..f791c54 100644
  #define MBEDTLS_X509_USE_C
  
  #define MBEDTLS_ASN1_PARSE_C
-@@ -100,7 +110,9 @@
+@@ -99,7 +108,9 @@
  #define MBEDTLS_ASN1_WRITE_C
  #define MBEDTLS_ECDSA_C
  #define MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
diff --git a/security/oc_tls.c b/security/oc_tls.c
index 0b7e5e59..a10f7531 100644
--- a/security/oc_tls.c
+++ b/security/oc_tls.c
@@ -94,30 +94,26 @@ static mbedtls_ssl_config client_conf[1];
 #define SHA256_MAC_KEY_LENGTH (32)
 #define SHA384_MAC_KEY_LENGTH (48)
 
-static const int ciphers[12] = {
+static const int ciphers[10] = {
   MBEDTLS_TLS_RSA_WITH_AES_256_CBC_SHA256,
   MBEDTLS_TLS_RSA_WITH_AES_128_GCM_SHA256,
   MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
   MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8,
   MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_128_CCM,
   MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,
-  //MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384,
-  MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
   MBEDTLS_TLS_ECDHE_PSK_WITH_AES_128_CBC_SHA256,
   MBEDTLS_TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,
   MBEDTLS_TLS_ECDH_ANON_WITH_AES_128_CBC_SHA256,
   0
 };
 #ifdef OC_CLIENT
-static const int anon_ciphers[12] = {
+static const int anon_ciphers[10] = {
   MBEDTLS_TLS_RSA_WITH_AES_256_CBC_SHA256,
   MBEDTLS_TLS_RSA_WITH_AES_128_GCM_SHA256,
   MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
   MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8,
   MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_128_CCM,
   MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,
-  //MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384,
-  MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
   MBEDTLS_TLS_ECDHE_PSK_WITH_AES_128_CBC_SHA256,
   MBEDTLS_TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,
   MBEDTLS_TLS_ECDH_ANON_WITH_AES_128_CBC_SHA256,
-- 
2.16.1.windows.1

